packages/nimble-components/src/rich-text-editor/index.ts

```diff
@@ -163,7 +205,100 @@ export class RichTextEditor extends FoundationElement {
+         * Assigned updated node in place of orderedList node from defaultMarkdownSerializer
+         * https://github.com/ProseMirror/prosemirror-markdown/blob/b7c1fd2fb74c7564bfe5428c7c8141ded7ebdd9f/src/to_markdown.ts#L94C2-L101C7
+         */
+        const orderedListNode = function orderedList(
+            state: MarkdownSerializerState,
+            node: Node
+        ): void {
+            const start = (node.attrs.start as number) || 1;
+            const maxW = String(start + node.childCount - 1).length;
+            const space = state.repeat(' ', maxW + 2);
+            state.renderList(node, space, i => {
+                const nStr = String(start + i);
+                return `${state.repeat(' ', maxW - nStr.length) + nStr}. `;
+            });
+        };
+
+        /**
+         * Internally Tiptap editor creates it own schema ( Nodes AND Marks ) based on the extensions ( Here Starter Kit is used for Bold, italic, orderedList and
+         * bulletList extensions) and defaultMarkdownSerializer uses schema from prosemirror-markdown to serialize the markdown.
+         * So, there is variations in the nodes and marks name (Eg. 'ordered_list' in prosemirror-markdown schema whereas 'orderedList' in tip tap editor schema),
+         * To fix up this reassigned the respective nodes and marks with tip-tap editor schema.
+         */
+        const nodes = {
+            bulletList: defaultMarkdownSerializer.nodes.bullet_list!,
+            listItem: defaultMarkdownSerializer.nodes.list_item!,
+            orderedList: orderedListNode,
+            doc: defaultMarkdownSerializer.nodes.doc!,
+            paragraph: defaultMarkdownSerializer.nodes.paragraph!,
+            text: defaultMarkdownSerializer.nodes.text!
+        };
+        const marks = {
+            italic: defaultMarkdownSerializer.marks.em!,
+            bold: defaultMarkdownSerializer.marks.strong!
+        };
+        return new MarkdownSerializer(nodes, marks);
+    }
+
+    /** This function takes a markdown string, parses it using the ProseMirror MarkdownParser, serializes the parsed content into a
+     * DOM structure using a DOMSerializer, and returns the serialized result.
+     * If the markdown parser returns null, it will clear the editor component by creating an empty document fragment.
+     */
+    private parseMarkdownToDOM(value: string): HTMLElement | DocumentFragment {
+        const parsedMarkdownContent = this.markdownParser.parse(value);
+        if (parsedMarkdownContent === null) {
+            return document.createDocumentFragment();
+        }
+
+        return this.domSerializer.serializeFragment(
+            parsedMarkdownContent.content
+        );
+    }
+
     private initializeEditor(): void {
+        // Creating div element to land tiptap editor
+        this.editor = document.createElement('div');
+        this.editor.className = 'editor';
+        this.editor.ariaMultiLine = 'true';
```

Comment:
The aria* properties don't work on firefox, you'll need to do `setAttribute('aria-multiline', 'true')` instead

Reply:
Thanks for your input. We have updated to `setAttribute` for setting the aria attribute and the test case passed.